index.d.ts

```diff
@@ -39,6 +46,15 @@ export interface Hooks {
 	*/
 	beforeRequest?: BeforeRequestHook[];
 
+	/**
+	Before the request is retried.
+
+	This hook enables you to modify the request right before retry. The hook function receives the `input`, `options`, `error` and `retryCount` arguments. You could, for example, modify `options.headers` here.
```

Comment:
This should have the exact same text and example as the readme. This makes for a better TS experience, and having them in sync makes it easier to maintain both.

// @szmarczak

Reply:
Looks like there's more stuff that's out of sync: https://github.com/sindresorhus/ky/commit/8ab3886a4b57ccf8e6c2ac55a8046c82f3ec211d

Reply:
Sorry, forgot about syncing these. There should be a tool that converts automatically TS docs into markdown, so we don't have to update the same info twice.

Reply:
Ideally, an GitHub Action, but for now, we just have to try to make sure all PRs update index.d.ts with both methods and docs and examples.

Reply:
I think a PR template with a checklist would be a good idea until it can be automated.